d2ast/d2ast.go

```diff
@@ -724,6 +724,12 @@ func (mk1 *Key) D2OracleEquals(mk2 *Key) bool {
 }
 
 func (mk1 *Key) Equals(mk2 *Key) bool {
+	if mk1 == nil && mk2 == nil {
+		return true
+	}
+	if (mk1 == nil) || (mk2 == nil) {
+		return false
+	}
```

Comment:
`if (mk1 == nil) != (mk2 == nil)`

Reply:
if both are nil we don't want to continue below which assumes != nil

Reply:
oh true